### DISCUSSION TITLE
How to get the editor instance inside a node.

### DISCUSSION THREAD

**Discussion by sifear** [facebook/lexical#4995] [Q&A] [answered]
I am trying to implement an ImageNode that saves and loads images from an object store (S3).
I call my method that downloads the image in the constructor function, but then, when i'd update the src prop with the ObjectURL, i get an error when calling my setter:

```
export class ImageNode extends DecoratorNode<JSX.Element> {
  constructor(...) {
    super(key);
    ...
    this.downloadFromS3();
  }
  setSrc(src: string) {
    const writable = this.getWritable();
    writable.__src = src;
  }
  ...
  async downloadFromS3(): Promise<void> {
    const src = await downloadImage({ ... });
    this.setSrc(src);
  }
}
```

This results in the error below:
`Error: Unable to find an active editor state. State helpers or node methods can only be used synchronously during the callback of editor.update() or editorState.read().`

My question is that, how to get an editor state inside my ImageNode. I couldn't find an example.
I appreciate any help here.

Edit: 
i wonder if this  should be the responsibility if ImageComponent, which uses the 
`const [editor] = useLexicalComposerContext();`
hook. Especially, that the image src is an object url and should not be serialized/de-serialized, but rather generated on upload and download.

**Comment by yf-yang** [comment, marked as answer]
https://lexical.dev/docs/concepts/editor-state#updating-state
You should instantiate the ImageNode class in an `editor.update` function call.
